### ISSUE TITLE
Add config-file for PHP-symbols

### ISSUE THREAD

**Issue by heiglandreas** [maglnet/ComposerRequireChecker#7]
Currently the internal symbol-table of the running PHP-binary is used to check for PHP-symbols that might be required. What happens when you required f.e. ```ext-ldap``` and run the checker using a PHP-binary that hasn't ext-ldap available? It will surely result in a false-positive.

I still need to check what happens but I'll report back.

Solution would be to have a config-file that contains **all** symbols that *might* be available in a PHP-build. This file could be created from the PHP-source code. That file would only contain the PHP-src and ex folders. PECL should **not** be included!

**Comment by maglnet** [comment]
Currently it throws an exception, we should find another solution than failing for this point :)
https://github.com/maglnet/ComposerRequireChecker/blob/faf96659320dc0ec3b7d2c8619732988d7ca063b/src/ComposerRequireChecker/DefinedSymbolsLocator/LocateDefinedSymbolsFromExtensions.php#L29

**Comment by maglnet** [comment]
I just realized, if you already require `ext-ldap` and it is not installed, a composer install should already fail.

But nevertheless, do you have any idea on how one could best create such a file from php-src?

**Comment by heiglandreas** [comment]
Well, you can always use ```composer install/update --ignore-platform-reqs```.

Excerpt from [the composer-page](https://getcomposer.org/doc/03-cli.md#install):

     ignore php, hhvm, lib-* and ext-* requirements and force the installation 
     even if the local machine does not fulfill these. See also the platform config option.

So it might happen.

The symbols can be filtered using something like 

    git clone https://github.com/php/php-src.git
    cd php-src
    grep -r PHP_FUNCTION ext | awk -F "/" '{print $2 ":" $3}' | awk -F ":" '{print $1 " " $3}' | sed "s/PHP_FUNCTION//" | sed "s/static //"

I'm pretty sure that the ```grep```-line can be optimized beyond my wildest dreams but it's a start!

**Comment by maglnet** [comment]
Right, didn't think about `--ignore-platform-reqs`.

The `grep` is a good start, I'll see if I can adjust it to fetch constants, classes and interfaces, too.

**Comment by heiglandreas** [comment]
```clang``` might also be a possibility to extract the informations. But I'm not deep enough into it at the moment. I need a bit more time for that. Might be the easiest to use the ```grep``` and a bit of manual adaption and then use ```clang``` for the next PHP-version.

**Comment by heiglandreas** [comment]
I've set up a rudimentary parser for the sources. 

https://github.com/maglnet/ComposerRequireChecker/compare/master...heiglandreas:feature/addSourceParser?expand=1

**Comment by maglnet** [comment]
@pavarnos I think your problem can already be solved by using an additional config file as described here: https://github.com/maglnet/ComposerRequireChecker#configuration

You may copy this file, add the symbols (function names and constant names) to the `symbol-whitelist` and then call comoposer-require-checker with the `--config-file=path/to/config.json` argument.

**Comment by pavarnos** [comment]
thanks. i did see that but wondered about overriding all these defaults with my own. Is there a way to _add_ to the defaults (in case they change in future)?

**Comment by maglnet** [comment]
There's #72 addressing exactly your issue. For now only overriding is possible.
